### DISCUSSION TITLE
Compilation using Docker

### DISCUSSION THREAD

**Discussion by anatmiller** [roc-streaming/roc-toolkit#790] [Discuss]
Hi, I am new here :) I just wanted to make sure. is the compilation on docker with Linux OS? are there any instructions on how to run that I missed.:) 
 thanks Anna

**Comment by gavv** [comment]
Hi! I don't quite understand the question.

We provide docker images with cross-compilation toolchains, which you can use on your Linux PC (x86_64) to build binaries for your Raspberry Pi or other Linux-based device (typically ARM or MIPS). See here: https://roc-streaming.org/toolkit/docs/building/user_cookbook.html

Docker images exist just for convenience, you can set up your own cross-compilation environment, see here: https://roc-streaming.org/toolkit/docs/portability/cross_compiling.html

Native compilation (when you build and run on the same PC) is not so complicated, so we don't provide any docker images for this case.

**Comment by anatmiller** [comment]
Thank you for your response! I actually managed to resolve the issue before your reply and successfully ran the code using Docker. I appreciate your detailed explanation
